### PR TITLE
 Merging PR #10215 into the next branch. 

### PR DESCRIPTION
**Pull request by NicholasCouri**
Merging PR #10215 into the next branch.

### CODE REVIEW

**Comment by tylerbutler** [comment]
@NicholasCouri Thank you!

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -858,28 +858,6 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
-            // This will patch snapshot in place!
-            // If storage is provided, it will wrap storage with BlobAggregationStorage that can
-            // pack & unpack aggregated blobs.
-            // Note that if storage is provided later by loader layer, we will wrap storage in this.storage getter.
-            // BlobAggregationStorage is smart enough for double-wrapping to be no-op
-            if (context.attachState === AttachState.Attached) {
-                // IContainerContext storage api return type still has undefined in 0.39 package version.
-                // So once we release 0.40 container-defn package we can remove this check.
-                assert(context.storage !== undefined, 0x1f4 /* "Attached state should have storage" */);
-                const aggrStorage = BlobAggregationStorage.wrap(
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:871]: This has to be removed from the `if (pendingRuntimeState)` block below as well.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -858,28 +858,6 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const baseSnapshot: ISnapshotTree | undefined = pendingRuntimeState?.baseSnapshot ?? context.baseSnapshot;
 
         let storage = context.storage;
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:860]: This assignment should check `pendingRuntimeState` and either use this directly or wrap around `SerializedSnapshotStorage`

Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:860]: let me know what you think.

**Comment by NicholasCouri** [comment]
np! :) 
:smile: 

---
In reply to: [1138858805](https://github.com/microsoft/FluidFramework/pull/10442#issuecomment-1138858805) [](http://example.com/codeflow?ancestors=1138858805,1138858805)

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 2 above)
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/containerRuntime.ts]: looks good.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -997,25 +963,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
-            this._storage = BlobAggregationStorage.wrap(
-                this.context.storage,
-                this.logger,
-                undefined /* allowPacking */,
-                this.disableIsolatedChannels ? 1 : 2,
-            );
-        }
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return this._storage!;
+        return this.context.storage;
```
Comment (wes-carlson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:966]: This needs to be this._storage I think

Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/containerRuntime.ts:966]: Yeah, storage is not just context.storage anymore. So, it should be passed to container runtime during construction.

Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:966]: We ended up removing the _storage when we removed the BlobAggregation usage (we always use context.storage now, I think)  - Is there a problem, you think ?

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 4]
```diff
@@ -856,42 +856,9 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
-                    context.storage,
-                    logger,
-                    undefined /* allowPacking */,
-                    packingLevel,
-                );
-            }, pendingRuntimeState.snapshotBlobs);
-        }
+        const storage = !pendingRuntimeState ?
+            context.storage :
+            new SerializedSnapshotStorage(() => { return context.storage; }, pendingRuntimeState.snapshotBlobs);
```
Comment (wes-carlson) [inline packages/runtime/container-runtime/src/containerRuntime.ts:861]: wrapping here has no effect if we just return context.storage in the getter

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 3 above)
Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:966]: Actually I think we would always use container.storage, isn't it ?

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 4 above)
Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:861]: Hey wes, SerializedSnapshotStorage  gets this as the first argument: " storageGetter: () => IDocumentStorageService,"

**Review by wes-carlson — APPROVED** [review]
:shipit:

**Review by kian-thompson — APPROVED** [review]

**Pull request merged**